src/Middleware/RequestThrottling/src/Internal/LoggerExtensions.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.RequestThrottling.Internal
+{
+    internal static class LoggerExtensions
+    {
+        private static readonly Action<ILogger, Exception> _requestEnqueued;
+        private static readonly Action<ILogger, Exception> _requestDequeued;
+
+        static LoggerExtensions()
+        {
+            _requestEnqueued = LoggerMessage.Define(
+                logLevel: LogLevel.Debug,
+                eventId: new EventId(1, "Request Enqueued"),
+                formatString: "Server is busy; queuing request in middleware."
```

Comment:
I think logging the number of currently queued requests here makes sense as part of this PR.

Reply:
@davidfowl @Tratcher @anurse What's our feeling about including in the TraceIdentifier in the log message as a structure parameter? I think we should. I know this can be pulled out of the scope, but this doesn't help you if you use console logging in its default configuration like we often do in our templates and samples.

[KestrelTrace.cs](https://github.com/aspnet/AspNetCore/blob/01a87989364dea1b18743a9fe3757eccc220b2ca/src/Servers/Kestrel/Core/src/Internal/Infrastructure/KestrelTrace.cs#L61) includes a TraceIdentifier structure parameter in logs where it's possible and uses ConnectionId otherwise. This is even done for logging when the app starts and stops reading the request body, so it's not like all these logs are being done outside of the request log scope and that's the reason for doing this.

Reply:
No, need it’s already part of the scope.

Reply:
Should we be removing ConnectionId and TraceId parameters from the Kestrel traces made inside the connection/request scope? Technically, that's a breaking change, right?

Reply:
We can talk about fixing it for console logging so that it is on by default